js/client-chat.js

```diff
@@ -1444,7 +1445,21 @@
 						this.$chat.children().last().append(' <button name="toggleMessages" value="' + user + '" class="subtle"><small>(' + $messages.length + ' line' + ($messages.length > 1 ? 's' : '') + ' from ' + user + ' hidden)</small></button>');
 					}
 					break;
-
+				case 'hidelines':
+					if (Dex.prefs('nounlink')) return;
+					var user = toID(row[2]);
+					var $messages = $('.chatmessage-' + user);
+					if (!$messages.length) break;
+					$messages.find('a').contents().unwrap();
+					if (row[1] === 'hide' || row[1] === 'delete') {
+						$messages = this.$chat.find('.chatmessage-' + user);
+						if (!$messages.length) break;
+						var lineCount = parseInt(row[3], 10) || 0;
+						if (lineCount) $messages = $messages.slice(-lineCount);
+						$messages.hide().addClass('revealed').find('button').parent().remove();
+						if (row[1] === 'hide') this.$chat.children().last().append(' <button name="toggleMessages" value="' + user + '" class="subtle"><small>(' + $messages.length + ' line' + ($messages.length > 1 ? 's' : '') + ' from ' + user + ' hidden)</small></button>');
```

Comment:
I don't suppose you could grab the user group to show auth the unhide button?

Reply:
I thought that it would be easier to handle this server side, by sending `|delete|` to regular users and `|hide|` to staff. Handling auth on the server seems like a cleaner permission to me.

Reply:
Hm, I suppose that works too. I think client is "better" but I suppose old-client architecture sucks so it's easier to do it server-side, so I support your decision to do it server-side.